gateway/gateway.go

```diff
@@ -276,10 +305,31 @@ func (gw *Gateway) handleMessage(msg config.Message, dest *bridge.Bridge) []*BrM
 		if dest.Protocol == "api" {
 			msg.Channel = originchannel
 		}
+
+		if canonicalParentMsgID != "" {
+			parentMsgID := ""
+			if res, ok := gw.Messages.Get(canonicalParentMsgID); ok {
```

Comment:
This is code duplication, better to create a new function eg `getDestMsgID` which can also replace the code on line 265
```
msg.ParentID = getDestMsgID(dest.Name,dest.Protocol,channel.ID,canonicalParentMsgID)
if msg.ParentID == "" {
   msg.ParentID = canonicalParentMsgID
}
```

```
func getDestMsgID(name string, protocol string, channelID string, origID string) string {
	if res, ok := gw.Messages.Get(origmsg.ID); ok {
		IDs := res.([]*BrMsgID)
		for _, id := range IDs {
			// check protocol, bridge name and channelname
			// for people that reuse the same bridge multiple times. see #342
			if protocol == id.br.Protocol && name == id.br.Name && channelID == id.ChannelID {
				return id.ID
			}
		}
	}
	return ""
}
```